### ISSUE TITLE
Query parameters don't work in history mode

### ISSUE THREAD

**Issue by bluego-joe** [lulupointu/vrouter#113]
When in history mode, it appears that route path matching occurs using the URL-encoded path, including query parameters. That means when running the app below and navigating to /content?foo=bar, you receive the error `The url '/content%3Ffoo=bar' has no matching route`.

```dart
void main() {
  runApp(
    VRouter(
      debugShowCheckedModeBanner: false,
      mode: VRouterMode.history,
      routes: [
        VWidget(
          path: '/content',
          widget: Text('Hello, World!'),
        ),
      ],
    ),
  );
}
```

In hash mode, this issue does not exist.

This is using version 1.2.0.

**Comment by lulupointu** [comment]
Thanks I'll have a look

**Comment by lulupointu** [comment]
I can reproduce this. I'll tell you when it's fixed.

Thanks for the nicely written issue btw :wink:

**Comment by lulupointu** [comment]
Fixed in `v1.2.0+5`.

I also added test so that this issue don't get into future updates

**Comment by bluego-joe** [comment]
Yep, looks fixed to me also :+1: 

Thanks for your fast response :)
